go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -32,6 +48,818 @@ func (c *Controller) processNextANPWorkItem(wg *sync.WaitGroup) bool {
+
+	var addrSet addressset.AddressSet
+	var err error
+	peerPodIPs := sets.Set[string]{}
+	for _, peer := range peers {
+		// TODO: Double check how empty selector means all labels match works
+		namespaces, err := c.anpNamespaceLister.List(peer.namespaceSelector)
+		if err != nil {
+			return nil, nil, err
+		}
+		namespaceCache := make(map[string]map[string]bool)
+		// NOTE: Multiple peers may match on same podIP which is fine, we use sets to store them to avoid duplication
+		for _, namespace := range namespaces {
+			podCache, ok := namespaceCache[namespace.Name]
+			if !ok {
+				podCache = make(map[string]bool)
+				namespaceCache[namespace.Name] = podCache
+			}
+			podNamespaceLister := c.anpPodLister.Pods(namespace.Name)
+			pods, err := podNamespaceLister.List(peer.podSelector)
+			if err != nil {
+				return nil, nil, err
+			}
+			for _, pod := range pods {
+				// we don't handle HostNetworked or completed pods; unscheduled pods shall be handled via pod update path
+				if util.PodWantsHostNetwork(pod) || util.PodCompleted(pod) || !util.PodScheduled(pod) {
+					continue
+				}
+				podIPs, err := util.GetPodIPsOfNetwork(pod, &util.DefaultNetInfo{})
+				if err != nil {
+					return nil, nil, err
+				}
+				peerPodIPs.Insert(ipsToString(podIPs)...)
+				podCache[pod.Name] = true
+			}
+		}
+		peer.namespaces = namespaceCache
+	}
+	asIndex := GetANPPeerAddrSetDbIDs(anpName, gressPrefix, fmt.Sprintf("%d", priority), controllerName, idType)
+	addrSet, err = c.addressSetFactory.EnsureAddressSet(asIndex)
+	if err != nil {
+		return nil, nil, err
+	}
+	return addrSet, peerPodIPs, nil
+}
+
+// getPortsOfSubject calculates all the LSP's that match for the provided anp's subject and returns them
+// It also populates the adminNetworkPolicySubject.namespaces and adminNetworkPolicySubject.podPorts
+// pieces of the cache
+func (c *Controller) getPortsOfSubject(anpSubject *adminNetworkPolicySubject) ([]*nbdb.LogicalSwitchPort, error) {
+	ports := []*nbdb.LogicalSwitchPort{}
+	anpSubject.podPorts = sets.Set[string]{}
+	namespaces, err := c.anpNamespaceLister.List(anpSubject.namespaceSelector)
+	if err != nil {
+		return nil, err
+	}
+	namespaceCache := make(map[string]map[string]bool)
+	for _, namespace := range namespaces {
+		podCache, ok := namespaceCache[namespace.Name]
+		if !ok {
+			podCache = make(map[string]bool)
+			namespaceCache[namespace.Name] = podCache
+		}
+		podNamespaceLister := c.anpPodLister.Pods(namespace.Name)
+		pods, err := podNamespaceLister.List(anpSubject.podSelector)
+		if err != nil {
+			return nil, err
+		}
+		for _, pod := range pods {
+			if util.PodWantsHostNetwork(pod) || util.PodCompleted(pod) || !util.PodScheduled(pod) {
+				continue
+			}
+			logicalPortName := util.GetLogicalPortName(pod.Namespace, pod.Name)
+			lsp := &nbdb.LogicalSwitchPort{Name: logicalPortName}
+			lsp, err = libovsdbops.GetLogicalSwitchPort(c.nbClient, lsp)
+			if err != nil {
```

Comment:
yeah the problem is I do :D to reduce noise there on the pod side, I end up only considering "important events" so i saw in the test that if i return nil here instead of error it was messing up the ordering.
but one on side, anp create/update should be rare, so perhaps its not so bad to have this retry  here than having to open up pod watcher to be more lavish,,,

Reply:
I see that pod handler only reacts when podIPs change, and it gets ips from the annotations, but the annotations are set by the pod handler before lsp is created, so you may get ips change event before lsp is created in the db.
So maybe we need one more condition, e.g. pod status (we know that pod can't be ready without lsp being created)

Reply:
> I see that pod handler only reacts when podIPs change, and it gets ips from the annotations, but the annotations are set by the pod handler before lsp is created, so you may get ips change event before lsp is created in the db. 

exactly, this is the problem.

> So maybe we need one more condition, e.g. pod status (we know that pod can't be ready without lsp being created)

oh really? status of pod reflects whether LSP was created or not? we will need to check other pod lifecycle events as well to see how much noise watching for pod.Status is going to give us..

Reply:
bring it up in the team meeting.

Reply:
going ahead with the `status.Running` check, it is a bit tricky to reproduce this gap between the annotation add v/s LSP add but for now let's do this.. if someone complains later on we can fix it propely when we fix the actual bug of "time of applying the policy" being too big between when pod is ready and that leading to traffic leak...